### DISCUSSION TITLE
How to handle Domain Validation/Exceptions?

### DISCUSSION THREAD

**Discussion by utkugenel** [ardalis/CleanArchitecture#715] [Q&A] [answered]
Example is not a real case, just to ask my question.

Error Response 1: ( fast-endpoints & Fluent validation)
`{
  "statusCode": 400,
  "message": "One or more errors occurred!",
  "errors": {
    "username": [
      "The length of 'Username' must be at least 2 characters. You entered 1 characters."
    ]
  }
}`

Error Response 2: (Argument Exception thrown from domain entity with extended ardalis GuardClauses)
`{
  "status": "Internal Server Error!",
  "code": 500,
  "reason": "Invalid username! Max length:15. Can only contain letters, numbers, hyphens (-) and underscores (_)",
  "note": "See application log for stack trace."
}`

Two different error response here. I want error messages in the same format.
What is best way to handle exceptions on domain?

**Comment by utkugenel** [comment, marked as answer]
I found my answer.
We can customize fast-endpoints's exception handler.
[https://fast-endpoints.com/docs/exception-handler#customizing-the-exception-handler](https://fast-endpoints.com/docs/exception-handler#customizing-the-exception-handler)

**Reply by ardalis** [reply]
See also:

https://youtu.be/dpPcnAT7n7M?si=Fy9sQUL8bALO6OZ6
